src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -57,478 +59,293 @@ public ResultTextMarker(int runId, Region region, string fullFilePath)
             }
-        }
 
-        private bool IsTracking()
-        {
-            return _docCookie.HasValue && IsTracking(_docCookie.Value);
-        }
+            if (vsTextView.GetBuffer(out IVsTextLines vsTextLines) != VSConstants.S_OK)
+            {
+                return false;
+            }
 
-        private void SubscribeToCaretEvents(IWpfTextView textView)
-        {
-            if (textView != null)
+            if (vsTextLines.GetLastLineIndex(out int lastLine, out int lastIndex) != VSConstants.S_OK)
             {
-                textView.Caret.PositionChanged += CaretPositionChanged;
-                textView.LayoutChanged += ViewLayoutChanged;
+                return false;
             }
-        }
 
-        private void ViewLayoutChanged(object sender, TextViewLayoutChangedEventArgs e)
-        {
-            if (_textView != null)
+            if (textSpan.iEndLine > lastLine)
             {
-                // If a new snapshot wasn't generated, then skip this layout
-                if (e.NewViewState.EditSnapshot != e.OldViewState.EditSnapshot)
-                {
-                    UpdateAtCaretPosition(_textView.Caret.Position);
-                }
+                textSpan.iEndLine = lastLine;
             }
-        }
 
-        private void CaretPositionChanged(object sender, CaretPositionChangedEventArgs e)
-        {
-            UpdateAtCaretPosition(e.NewPosition);
-        }
+            // Now fix up the column numbers.
+            bool coerced = false;
+            ITextSnapshotLine startTextLine = textSnapshot.GetLineFromLineNumber(textSpan.iStartLine);
 
-        private void UpdateAtCaretPosition(CaretPosition caretPoisition)
-        {
-            // Check if the current caret position is within our region. If it is, raise the RegionSelected event.
-            if (_trackingSpan.GetSpan(_trackingSpan.TextBuffer.CurrentSnapshot).Contains(caretPoisition.Point.GetPoint(_trackingSpan.TextBuffer, PositionAffinity.Predecessor).Value))
+            if (textSpan.iStartIndex < 0 || textSpan.iStartIndex >= startTextLine.Length)
             {
-                OnRaiseRegionSelected(new EventArgs());
+                textSpan.iStartIndex = 0;
+                coerced = true;
             }
-        }
 
-        protected virtual void OnRaiseRegionSelected(EventArgs e)
-        {
-            EventHandler handler = RaiseRegionSelected;
+            ITextSnapshotLine endTextLine = textSnapshot.GetLineFromLineNumber(textSpan.iEndLine);
 
-            if (handler != null)
+            // If we are highlighting just one line and the column values don't make
+            // sense or we corrected one or more of them, then simply mark the
+            // entire line
+            if (textSpan.iEndLine == textSpan.iStartLine && (coerced || textSpan.iStartIndex >= textSpan.iEndIndex))
             {
-                handler(this, e);
+                textSpan.iStartIndex = 0;
```

Comment:
Weird if coerced, shouldn't iStartIndex already be zero?

Reply:
It really can't, but it does make it clear that the entire line is being selected. (This was existing code, so leaving it)

---
In reply to: [481398586](https://github.com/microsoft/sarif-visualstudio-extension/pull/191#discussion_r481398586) [](ancestors = 481398586)